camel/models/azure_openai_model.py

```diff
@@ -0,0 +1,120 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from typing import Any, Dict, List, Optional, Union
+
+from openai import AzureOpenAI, Stream
+
+from camel.configs import OPENAI_API_PARAMS_WITH_FUNCTIONS
+from camel.messages import OpenAIMessage
+from camel.models import BaseModelBackend
+from camel.types import ChatCompletion, ChatCompletionChunk, ModelType
+from camel.utils import BaseTokenCounter, OpenAITokenCounter
+
+azure_model_name_to_type = {
+    "gpt-35-turbo": ModelType.GPT_3_5_TURBO,
+    "gpt-35-turbo-16k": ModelType.GPT_3_5_TURBO_16K,
+    "gpt-4": ModelType.GPT_4,
+    "gpt-4-32k": ModelType.GPT_4_32K,
+    "gpt-4-1106-preview": ModelType.GPT_4_TURBO,
+}
```

Comment:
What is this used for?

Reply:
Code for an older implementation, will be remove in the next commit.